test/base/utils/TestITensorsExportedNames/TestITensorsExportedNames.jl

```diff
@@ -8,17 +8,186 @@ open("itensors_exported_names.jl", "w") do io
+  :TagSet, # remove export
+  :addblock!, # deprecate
+  :addtags,
+  :addtags!, # deprecate
+  :allhastags, # deprecate
+  :anyhastags, # deprecate
+  :array, # deprecate
+  :axpy!, # remove export
+  :blockdim, # deprecate
+  :blockoffsets, # deprecate
+  :checkflux,
+  :combinedind,
+  :combiner,
+  :commonind,
+  :commonindex, # deprecate
+  :commoninds,
+  :complex!, # deprecate
+  :contract,
+  :convert_eltype, # remove export
+  :dag,
+  :delta,
+  :dense,
+  :denseblocks,
+  :diag,
+  :diagITensor,
+  :diagitensor, # deprecate
+  :dim, # remove export
+  :dims, # remove export
+  :dir,
+  :directsum,
+  :disable_combine_contract!, # deprecate
+  :disable_tblis!, # deprecate
+  :disable_warn_order!, # remove export
+  :eachindval, # deprecate
+  :eachnzblock, # deprecate
+  :eachval, # deprecate
+  :eigen, # remove export
+  :emptyITensor, # deprecate
+  :enable_combine_contract!, # deprecate
+  :enable_tblis!, # deprecate
+  :factorize, # remove export
+  :filterinds,
+  :findindex, # deprecate
+  :findinds, # deprecate
+  :firstind,
+  :firstintersect, # deprecate
+  :firstsetdiff, # deprecate
+  :flux,
+  :fparity, # remove export
+  :getfirst, # remove export
+  :getindex, # remove export
+  :hadamard_product, # remove export
+  :hascommoninds,
+  :hasid,
+  :hasind,
+  :hasinds,
+  :hasplev,
+  :hasqns,
+  :hastags,
+  :id,
+  :ind,
+  :index_id_rng, # remove export
+  :inds,
+  :insertblock!, # remove export
+  :isactive, # remove export
+  :isfermionic, # remove export
+  :ishermitian, # remove export
+  :isindequal, # remove export
+  :itensor,
+  :lq, # remove export
+  :mapprime,
+  :mapprime!, # remove export (not defined?)
+  :matmul, # deprecate
+  :matrix, # deprecate
+  :modulus, # remove export
+  :mul!, # remove export
+  :nblocks,
+  :nnz,
+  :nnzblocks,
+  :noncommonind,
+  :noncommoninds,
+  :noprime,
+  :noprime!, # deprecate
+  :norm, # remove export
+  :normalize, # remove export
+  :normalize!, # remove export
+  :not, # remove export
+  :nullspace,
+  :nzblock,
+  :nzblocks,
+  :onehot,
+  :order,
+  :permute,
+  :plev,
+  :polar,
+  :pop,
+  :popfirst,
+  :prime,
+  :prime!,
+  :push,
+  :pushfirst,
+  :ql,
+  :qn,
+  :qr,
+  :randn!, # remove export
+  :readcpp,
+  :removeqn,
+  :removeqns,
+  :removetags,
+  :removetags!, # deprecate
+  :replaceind,
+  :replaceind!,
+  :replaceindex!,
+  :replaceinds,
+  :replaceinds!,
+  :replacetags,
+  :replacetags!,
+  :reset_warn_order!,
+  :rmul!,
+  :rq,
+  :scalar,
+  :scale!,
+  :set_warn_order!,
+  :setdir,
+  :setelt,
+  :setindex,
+  :setprime,
+  :setprime!,
+  :setspace,
+  :settags,
+  :settags!,
+  :sim,
+  :sim!,
+  :space,
+  :storage,
+  :store,
+  :svd,
+  :swapind,
+  :swapinds,
+  :swapinds!,
+  :swapprime,
+  :swapprime!,
+  :swaptags,
+  :swaptags!,
+  :tags,
+  :transpose,
+  :unionind,
+  :unioninds,
+  :uniqueind,
+  :uniqueindex,
+  :uniqueinds,
+  :use_combine_contract,
+  :use_debug_checks,
+  :vector,
+  :δ,
+  :⊕,
+  :⊙,
+]
```

Comment:
@emstoudenmire please go through this list and see if there is anything we can argue could get moved to the `ITensorMPS.jl` export list, which would mean they would get removed from the `ITensors.jl` export list once we move `ITensors.ITensorMPS` to `ITensorMPS.jl`.

Many of them should be, or already are, deprecated, and I started to comment about those, but that is a separate issue. We should still try to categorize them by ones that are more associated with MPS operations vs. more general ITensor operations. Note that we may remove some things from the `ITensors.jl` export list, like `OpSum`, which we will keep in `ITensors.jl` since other packages like `ITensorNetworks.jl` will be using them.

Reply:
Good question... Looking over the above list a few times, I did not see any that jumped out at me as being "MPS only". Some like `replacetags` certain have notable MPS/MPO overloads, but also make sense purely at the ITensor level. Were there any specifically you were still undecided about in terms of which package they should get exported from?

I do agree that reducing this list and carrying through any deprecations or planned deprecations is a great idea, with the aim of reducing the exports.

Reply:
The main ones I was most uncertain about were the ones that were not obviously ITensor or MPS functionality, i.e. functions that you brought up below like `state`, `siteind[s]`, `op`, etc. which can be used with either ITensor operations or MPS/MPO operations but are more often used in association with MPS algorithms in practice. I'm biasing towards moving them to `ITensorMPS` (at least their exports, for the time being their implementation will live in `ITensors.jl`) but it is helpful to get a second pair of eyes on that decision as a sanity check. Also it is a long list so I wanted to double check I wasn't making an obvious mistake somewhere.